sdks/python/src/opik/evaluation/models/litellm/litellm_chat_model.py

```diff
@@ -193,12 +286,25 @@ def generate_provider_response(
         ):
             all_kwargs = opik_monitor.try_add_opik_monitoring_to_params(all_kwargs)
 
-        response = self._engine.completion(
-            model=self.model_name, messages=messages, **all_kwargs
+        last_exception = None
+        for attempt in range(max_attempts):
```

Comment:
You can just use `tenacity` to decorate the call with the retry logic you need, let's not re-invent it manually with a risk to introduce a bug.